lib/grp.gi

```diff
@@ -339,6 +339,10 @@ InstallImmediateMethod( IsPerfectGroup,
     0,
     grp -> Size( grp ) = 1 );
 
+InstallMethod( IsPerfectGroup, "for groups having abelian invariants",
+    [ IsGroup and HasAbelianInvariants ],
+    grp -> Length( AbelianInvariants( grp ) ) = 0 );
+
```

Comment:
This function is not currently covered by the tests.

Indeed, I just looked into it and noticed that there is an almost identical method being installed for groups in the filter `IsSubgroupFpGroup` in grpf.gi, line 910.

The difference if of course that this method gets invoked even if they are not yet computed.

Which leaves me wondering how to create a group for which this method gets triggered...